### PR TITLE
Use `master` branches for parity dependencies

### PR DESCRIPTION
**Pull request by alistair-singh**
The current bridge hub work relies on `master` branches for `polkadot` and `substrate` dependencies.

The upgrade consists of:
1. Remove the deprecated `generate_store` macro. https://github.com/paritytech/substrate/issues/13260
2. Remove deprecated `Weights::from_ref_time`. https://github.com/paritytech/substrate/pull/13475

Snowfork Cumulus companion:  https://github.com/Snowfork/cumulus/pull/7

### CODE REVIEW

**Review by vgeddes — COMMENTED** [review]
Lets also update codec, serde, and scale-info deps to match Substrate master, where I'm seeing the following versions in use:

* codec: 3.2.2
* serde: 1.0.136
* scale-info: 2.1.1

**Comment by alistair-singh** [comment]
> Lets also update codec, serde, and scale-info deps to match Substrate master, where I'm seeing the following versions in use:

The current branch I have based this on `bridge-hub-rococo-wococo` has the following versions with the `bridge-hub-rococo` runtime.

* codec: 3.2.1
* serde: 1.0.156
* scale-info: 2.3.1

I have updated our pallets to match those versions instead. 
See https://github.com/Snowfork/cumulus/blob/alistair/use-rococo-bridge/parachains/runtimes/bridge-hubs/bridge-hub-rococo/Cargo.toml

File: parachain/pallets/xcm-support/Cargo.toml
Comment (yrong) [inline parachain/pallets/xcm-support/Cargo.toml:1]: Is this pallet still required or can we just removed it?  seems already coverd in xcm pallet
https://github.com/paritytech/polkadot/blob/1fe1702508f24b96b136f696fc180d6b27145c8b/xcm/pallet-xcm/src/lib.rs#L839-L869

cumulus already contains

Comment (alistair-singh) [reply parachain/pallets/xcm-support/Cargo.toml:1]: Addressed in https://github.com/Snowfork/snowbridge/pull/796/commits/5ced5d2243325aa1d38148eca77f043c03f7ad3e

**Review by yrong — APPROVED** [review]
LGTM!

**Review by vgeddes — APPROVED** [review]
+1

**Pull request merged**